### ISSUE TITLE
Incorrect warning of required parameter when using array that includes variable.

### ISSUE THREAD

**Issue by thlapin** [rodjek/puppet-lint#537]
Example:

```
# class doc
class test (
  $var1 = [$::hostname, 'host'],
) {
}
```

Results in
`WARNING: optional parameter listed before required parameter on line 3`
